Describe the new filter operations (#1055)
DevExpress/devextreme-documentation: api-reference/_hidden/GridBaseColumn/filterOperations.md

---
---
##### shortDescription

> Applies only if [allowFiltering](/api-reference/_hidden/GridBaseColumn/allowFiltering.md '{basewidgetpath}/Configuration/columns/#allowFiltering') and at least one of the [filterRow](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/filterRow '{basewidgetpath}/Configuration/filterRow/').**visible** and [filterPanel]({basewidgetpath}/Configuration/filterPanel/).**visible** options are **true**.

Reviewer: ```suggestion
Specifies available filter operations. Applies if [allowFiltering](/api-reference/_hidden/GridBaseColumn/allowFiltering.md '{basewidgetpath}/Configuration/columns/#allowFiltering') is **true** and the [filterRow](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/filterRow '{basewidgetpath}/Configuration/filterRow/'), [filterPanel]({basewidgetpath}/Configuration/filterPanel/), or both are visible.
```

> Specifies available filter operations. Applies if [allowFiltering](/api-reference/_hidden/GridBaseColumn/allowFiltering.md '{basewidgetpath}/Configuration/columns/#allowFiltering') is **true** and the [filterRow](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/filterRow '{basewidgetpath}/Configuration/filterRow/'), [filterPanel]({basewidgetpath}/Configuration/filterPanel/), or both are visible.

Reviewer 2: ```suggestion
Specifies the available filter operations. Applies if [allowFiltering](/api-reference/_hidden/GridBaseColumn/allowFiltering.md '{basewidgetpath}/Configuration/columns/#allowFiltering') is **true** and the [filterRow](/api-reference/10%20UI%20Widgets/GridBase/1%20Configuration/filterRow '{basewidgetpath}/Configuration/filterRow/') and/or [filterPanel]({basewidgetpath}/Configuration/filterPanel/) are visible.
```

---

> The following table lists available filters by [data types](/api-reference/_hidden/GridBaseColumn/dataType.md '{basewidgetpath}/Configuration/columns/#dataType') that can be used both in the filter row and nested [filterBuilder](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/#filterBuilder) component. The same filters are assigned to columns of a specific data type by default.

Reviewer: > that can be used both in the filter row and nested [filterBuilder](/Documentation/ApiReference/UI_Widgets/dxDataGrid/Configuration/#filterBuilder) component.

I didn't understand this part. Are data types can be used or filters? What do mean by "used"? Can we be more specific?

> The same filters are assigned to columns of a specific data type by default.

"By default" is an outlaw, we better rewrite this sentence while we're at it.

Author: I suggest the followng:

1. The following table lists filter operations by [data types](). Users can apply these filters both in the filter row and nested [filterBuilder]() component. 

I would say that use of "by default" was justified in the next sentence, because this phrase is much more concise than the description I can make up:

2. **DataGrid** automatically assigns these filters operations to columns of a specific data type.
2.1 My second thought was the following: 'When **DataGrid** is initialized, it automatically assigns...'

But both these sentences seem a bit clumsy to me. Could you please share your thoughts?

Reviewer: ```suggestion
The following table lists available filter operations by [data types](/api-reference/_hidden/GridBaseColumn/dataType.md '{basewidgetpath}/Configuration/columns/#dataType'). Users can apply these filter operations both in the filter row and nested [filterBuilder]({basewidgetpath}/Configuration/#filterBuilder) component. The same filters are assigned to columns of a specific data type by default.
```

> The following table lists available filter operations by [data types](/api-reference/_hidden/GridBaseColumn/dataType.md '{basewidgetpath}/Configuration/columns/#dataType'). Users can apply these filter operations both in the filter row and nested [filterBuilder]({basewidgetpath}/Configuration/#filterBuilder) component. The same filters are assigned to columns of a specific data type by default.

Reviewer: ```suggestion
The following table lists available filter operations by [data types](/api-reference/_hidden/GridBaseColumn/dataType.md '{basewidgetpath}/Configuration/columns/#dataType'). Users can apply these filter operations both in the filter row and nested [filterBuilder]({basewidgetpath}/Configuration/#filterBuilder) component. The same operations are assigned to columns of a specific data type by default.
```

> The following table lists available filter operations by [data types](/api-reference/_hidden/GridBaseColumn/dataType.md '{basewidgetpath}/Configuration/columns/#dataType'). Users can apply these filter operations both in the filter row and nested [filterBuilder]({basewidgetpath}/Configuration/#filterBuilder) component. The same operations are assigned to columns of a specific data type by default.

Reviewer 2: ```suggestion
The following table lists the available filter operations by [data types](/api-reference/_hidden/GridBaseColumn/dataType.md '{basewidgetpath}/Configuration/columns/#dataType'). Users can apply these filter operations in the filter row and nested [filterBuilder]({basewidgetpath}/Configuration/#filterBuilder) component. The same operations are assigned to columns of a specific data type.
```

  </tr>
</table>

> The following table contains filter operations specific for the nested filter builder:

Reviewer: As I understand, some filter operations can be used in both the filter row and the filter builder and other filter operations can be used in the filter builder only. If that is so, then this thought should be articulated more clearly.

Author: ```suggestion
The following table contains filters that users can apply only in the nested filter builder:
```

Reviewer: ```suggestion
The following table contains filter operations that users can apply only in the nested filter builder:
```

  </tr>
</table>

> The nested filter builder also allows users to select from an extended set of operations that includes *"anyof"*, *"noneof"*, *"isblank"*, *"isnotblank"*, and [names](/Documentation/ApiReference/UI_Widgets/dxFilterBuilder/Configuration/customOperations/#name) of custom operations (if there are any).

Reviewer 2: ```suggestion
The nested filter builder also allows users to select from an extended set of operations that include *"anyof"*, *"noneof"*, *"isblank"*, *"isnotblank"*, and [custom operations](/Documentation/ApiReference/UI_Widgets/dxFilterBuilder/Configuration/customOperations/#name).
```

Author: Hello. The point here is that the `filterOperations` array cannot contain custom operations, but only their names of the *string* type. The custom operations themselves should be implemented separately as follows: 
```
 customOperations: [{
                name: "weekends",
                calculateFilterExpression: function() {
                    // blah blah
                }
            }]
```

Then we take the value of the `name` field and pass it to the `filterOperations` array.

Reviewer 2: I see. Ok, use [name] then
